### PR TITLE
Upgrade redis client v8

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description

- Upgrade redis client library from v7 to v8
- Upgrade Go version to `1.15`

## Root Cause
The go-redis library version we use is outdated

## Solution
fix API changes, mostly v8 requires context to make Redis calls

## Testing
Both

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [x] You have added unit tests

### CODE REVIEW

File: iowrappers/redis_client.go [hunk 1]
```diff
@@ -26,6 +27,12 @@ const (
 	NumVisitorsPrefix      = "visitor_count"
 )
 
+var RedisClientContext context.Context
+
+func init() {
+	RedisClientContext = context.Background()
```
Comment (sananand007) [inline iowrappers/redis_client.go:33]: why use a `Background` and not any other type of contexts ?

**Review by sananand007 — APPROVED** [review]
lgtm. nit: wanted to understand why are we using context.background ?

File: iowrappers/redis_client.go [hunk 2]
```diff
@@ -4,10 +4,11 @@ package iowrappers
 // serving the caching needs of the Vacation Planner
 
 import (
+	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
-	"github.com/go-redis/redis/v7"
+	"github.com/go-redis/redis/v8"
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply iowrappers/redis_client.go:11]: Good question, they didn't provide a feature list though. But I think from API we can see they are making some changes like `context`

File: iowrappers/redis_client.go (hunk 1 above)
Comment (weihesdlegend) [reply iowrappers/redis_client.go:33]: because at the moment we are not doing anything except satisfying the function signature requirement

**Pull request merged**